Userland/Libraries/LibAudio/PulseAudioOutputImplementation.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023, Gregory Bertilson <zaggy1024@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PulseAudioOutputImplementation.h"
+
+namespace Audio {
+
+ErrorOr<NonnullOwnPtr<OutputImplementation>> PulseAudioOutputImplementation::create(OutputState initial_state, u32 sample_rate, u8 channels, u32 target_latency_ms, AudioDataRequestCallback const& data_request_callback)
+{
+    VERIFY(data_request_callback);
+    auto context = TRY(PulseAudioContext::instance());
+    auto stream = TRY(context->create_playback_stream(initial_state, sample_rate, channels, target_latency_ms, [&](PulseAudioPlaybackStream&, Bytes buffer, size_t sample_count) {
```

Comment:
Capturing `data_request_callback` by reference and using it in an async callback here feels quite unsafe. Let's move it in:
```c++
	auto stream_data_request_callback = [data_request_callback = move(data_request_callback)](PulseAudioPlaybackStream&, Bytes buffer, size_t sample_count) {
        return data_request_callback(buffer, PcmSampleFormat::Float32, sample_count);
    });

    auto stream = TRY(context->create_playback_stream(initial_state, sample_rate, channels, target_latency_ms, move(stream_data_request_callback));
```

Reply:
Ah, that was because I couldn't move a const capture when I was trying to figure out why the callback wasn't working initially, but if I change the thread's lambda in `PlaybackStream::create()` to a mutable one, it works correctly.